### DISCUSSION TITLE
When clicking "x" in the search bar, value of "q" is removed as expected, however the value of "f" remains, how can I have that cleared as well when 'x" is clicked?

### DISCUSSION THREAD

**Discussion by tedtninety** [microsoft-search/pnp-modern-search#3153] [Ideas]
I have a couple of direct links  ( on a nav , on the same page as the search box  ) with "q" and "f" parameters in them: "q" for the keyword inputted and "f" to filter a RefinableString.
When a user clicks "x" to clear the input, the "x" parameter is cleared, however the "f" parameter remains on the URL, and it is unfortunately applied to any subsequent input.
How can I make it so that when "'x"is clicked all parameters are cleared.
Thank you.

**Comment by wobba** [comment]
Sounds like an enhancement. As the search box web part don't have knowledge of the "f" parameter, assuming you use it in the result web part, we need to figure out how best to configure this to make it happen. Happy to have someone in the community propose a suggestions and maybe code it.

**Reply by tedtninety** [reply]
@wobba Thank you for your kind consideration.
